### PR TITLE
45 message card

### PR DESCRIPTION
**Pull request by fadeomar**
**45 message card**

### create the message card component that contain a container for all element and inside this container 
there is :

1. the sender message's photo as an avatar 
2. his/her name as a username .
3. the last message as a paragraph.
4. the time of last message as a span .

----
how to use it :
1.
```
    const userInfo = [
      {
        id: 5,
        username: 'Fatmasiam',
        email: 'fatma.o.siam@gmail.com',
        lastMessage: 'I\'m fine',
        avatar: null,
        messageTime: '11:15 ',
      },
      {
        id: 6,
        username: 'Fade',
        email: 'fade@gmail.com',
        lastMessage: 'how are you?',
        avatar: null,
        messageTime: '11:15 ',
      },
    ];
```
2. in app.js file  

- import UserCard from './components/commonComponents/UserCard'; 

- 
```
        {userInfo.map(info => (
          <MessageCard key={info.id} messageInfo={info} />
        ))}

```

note :+1: : 
 you must import the component before using it in the app file :1st_place_medal: 
```
import MessageCard from './components/commonComponents/MessageCard';
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,8 +36,9 @@
     "env2": "^2.2.2",
     "express": "^4.17.1",
     "jsonwebtoken": "^8.5.1",
-    "pg": "^7.12.1",
     "morgan": "^1.9.1",
+    "pg": "^7.12.1",
+    "prop-types": "^15.7.2",
```
Comment (Fatmasiam) [inline package.json]: Why you install the props-types here 😱
It's should be on the `client/package.json`

File: client/src/components/commonComponents/MessageCard/index.css [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+.message-time {
+  display: flex;
+  align-items: center;
+}
+
+.message-time span {
+  font-family: Roboto;
+font-style: normal;
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/commonComponents/MessageCard/index.css]: done @Fatmasiam

File: package.json (hunk 1 above)
Comment (fadeomar) [reply package.json]: delete it

File: client/src/components/commonComponents/MessageCard/index.css [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+font-weight: 800;
+font-size: 16px;
+line-height: 16px;
+display: flex;
+align-items: center;
+}
```
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/commonComponents/MessageCard/index.css]: done @Fatmasiam

File: package.json (hunk 1 above)
Comment (fadeomar) [reply package.json]: delete it

File: client/src/components/commonComponents/MessageCard/index.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (fadeomar) [reply client/src/components/commonComponents/MessageCard/index.css]: I added a new line put didn't appears at github

File: client/src/components/commonComponents/MessageCard/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './index.css';
+
+const MessageCard = ({
+  messageInfo: { avatar, username, lastMessage, messageTime },
```
Comment (Amoodaa) [inline client/src/components/commonComponents/MessageCard/index.js]: do the default img link in the props not with `||` maybe

File: client/src/components/commonComponents/MessageCard/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    <div className="message-time">
+      <span>{messageTime}</span>
+    </div>
+  </div>
+);
+
+MessageCard.propTypes = {
+  messageInfo: PropTypes.shape({
+    username: PropTypes.string.isRequired,
+    avatar: PropTypes.string.isRequired,
```
Comment (Amoodaa) [inline client/src/components/commonComponents/MessageCard/index.js]: you'll need to remove `isRequired` here

**Comment by Fatmasiam** [comment]
@fadeomar  look to these comments

**Pull request merged**